src/wallet/rpcwallet.cpp

```diff
@@ -1136,43 +1135,57 @@ UniValue getshieldedbalance(const JSONRPCRequest& request)
     if (!pwalletMain)
         return NullUniValue;
 
-    if (request.fHelp || request.params.size() > 2)
+    if (request.fHelp || request.params.size() > 3)
         throw std::runtime_error(
-                "getshieldedbalance ( minconf includeWatchonly )\n"
-                "\nReturn the total shielded value of funds stored in the node's wallet.\n"
+                "getshieldedbalance \"address\" ( minconf includeWatchonly )\n"
+                "\nReturn the total shielded value of funds stored in the node's wallet or if an address was given,"
+                "\nreturns the balance of the shielded addr belonging to the node's wallet.\n"
                 "\nCAUTION: If the wallet contains any addresses for which it only has incoming viewing keys,"
                 "\nthe returned private balance may be larger than the actual balance, because spends cannot"
                 "\nbe detected with incoming viewing keys.\n"
                 "\nArguments:\n"
-                "1. minconf          (numeric, optional, default=1) Only include private and transparent transactions confirmed at least this many times.\n"
-                "2. includeWatchonly (bool, optional, default=false) Also include balance in watchonly addresses (see 'importaddress' and 'z_importviewingkey')\n"
+                "1. \"address\"      (string) The selected address. It must be a Sapling address.\n"
+                "2. minconf          (numeric, optional, default=1) Only include private and transparent transactions confirmed at least this many times.\n"
+                "3. includeWatchonly (bool, optional, default=false) Also include balance in watchonly addresses (see 'importaddress' and 'z_importviewingkey')\n"
                 "\nResult:\n"
                 "amount              (numeric) the total balance of shielded funds (in Sapling addresses)\n"
                 "\nExamples:\n"
                 "\nThe total amount in the wallet\n"
-                + HelpExampleCli("getshieldedbalance", "") +
+                + HelpExampleCli("getshieldedbalance", "")
+                + HelpExampleCli("getshieldedbalance", "ptestsapling1h0w73csah2aq0a32h42kr7tq4htlt5wfn4ejxfnm56f6ehjvek7k4e244g6v8v3pgylmz5ea8jh") +
                 "\nThe total amount in the wallet at least 5 blocks confirmed\n"
-                + HelpExampleCli("getshieldedbalance", "5") +
+                + HelpExampleCli("getshieldedbalance", "\"*\" \"5\"") +
                 "\nAs a json rpc call\n"
-                + HelpExampleRpc("getshieldedbalance", "5")
+                + HelpExampleRpc("getshieldedbalance", "\"*\" \"5\"")
         );
 
     LOCK2(cs_main, pwalletMain->cs_wallet);
 
-    int nMinDepth = 1;
+    libzcash::PaymentAddress address{libzcash::InvalidEncoding()};
     if (request.params.size() > 0) {
-        nMinDepth = request.params[0].get_int();
+        std::string addressStr = request.params[0].get_str();
+        if (addressStr.empty() || addressStr != "*") {
```

Comment:
the `"*"` is purely a "show me all", the user can input an empty string or an "*" to call for the same behavior for now.

I would actually prefer to remove the empty space (if we really want to remove one of those in a future PR), call the command leaving the string empty is not helpful.
Think on: `getshieldedbalance   5` --> hard to know that there is an empty string in-between `getshieldedbalance` and 5 denoting the "show me all".

Reply:
> Think on: getshieldedbalance 5 --> hard to know that there is an empty string in-between getshieldedbalance and 5 denoting the "show me all".

Well, it would be an empty string... we still have positional arguments, so `getshieldedbalance """ 5`.
Anyway, this is surely secondary in this PR, and can be tackled later if need be.